eventhubs/eventhubs_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft and contributors.  All rights reserved.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package eventhubs
+
+import (
+	"context"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/Azure-Samples/azure-sdk-for-go-samples/helpers"
+	"github.com/Azure-Samples/azure-sdk-for-go-samples/iam"
+	"github.com/Azure-Samples/azure-sdk-for-go-samples/resources"
+)
+
+const (
+	location = "westus2"
+	nsName   = "ehtest-03-ns"
+	hubName  = "ehtest-03-hub"
+
+	// for storage.LeaserCheckpointer
+	storageAccountName   = "ehtest0001storage"
+	storageContainerName = "eventhubs0001leasercheckpointer"
+)
+
+func TestMain(m *testing.M) {
+	err := helpers.ParseArgs()
+	if err != nil {
+		log.Fatalf("failed to parse args: %v\n", err)
+	}
+
+	err = iam.ParseArgs()
+	if err != nil {
+		log.Fatalln("failed to parse IAM args")
+	}
+
+	os.Exit(m.Run())
+}
+
+func ExampleEventHubs() {
+	ctx := context.Background()
+	defer resources.Cleanup(ctx)
+
+	// create group
+	var err error
+	helpers.SetResourceGroupName("eventhubstest")
+	_, err = resources.CreateGroup(ctx, helpers.ResourceGroupName())
+	if err != nil {
+		helpers.PrintAndLog(err.Error())
+	}
+
+	// create Event Hubs namespace
+	_, err = CreateNamespace(ctx, nsName)
+	if err != nil {
+		helpers.PrintAndLog(err.Error())
+	}
+	helpers.PrintAndLog("created namespace")
+
+	// create Event Hubs hub
+	_, err = CreateHub(ctx, nsName, hubName)
+	if err != nil {
+		helpers.PrintAndLog(err.Error())
+	}
+	helpers.PrintAndLog("created hub")
+
+	// send and receive messages
+	log.Printf("Send()\n")
+	Send()
+	log.Printf("Receive()\n")
+	Receive()
+	log.Printf("ReceiveViaEPH()\n")
+	ReceiveViaEPH()
```

Comment:
Instead of creating a new context in `ReceiveViaEPH`, I'd pass in the one created here. :)

Reply:
Fair enough, working on this now, thank you for helping adopt better practices in the samples!